internal/engine/evaluator.go

```diff
@@ -114,278 +118,264 @@ func (noopEvaluator) Evaluate(_ context.Context, _ tracer.Context, _ *enginev1.C
 	return nil, ErrPolicyNotExecutable
 }
 
-type rolePolicyEvaluator struct {
-	policies map[string]*runtimev1.RunnableRolePolicySet
+type ruleTableEvaluator struct {
+	*ruletable.RuleTable
+	schemaMgr  schema.Manager
+	evalParams evalParams
 }
 
-func newRolePolicyEvaluator(rps []*runtimev1.RunnablePolicySet) *rolePolicyEvaluator {
-	policies := make(map[string]*runtimev1.RunnableRolePolicySet)
-	for _, p := range rps {
-		if rp, ok := p.PolicySet.(*runtimev1.RunnablePolicySet_RolePolicy); ok {
-			policies[rp.RolePolicy.Role] = rp.RolePolicy
-		}
+func (rte *ruleTableEvaluator) Evaluate(ctx context.Context, tctx tracer.Context, input *enginev1.CheckInput) (*PolicyEvalResult, error) {
+	version := input.Resource.PolicyVersion
+	if version == "" {
+		version = defaultVersion
 	}
 
-	return &rolePolicyEvaluator{policies: policies}
-}
+	request := checkInputToRequest(input)
```

Comment:
This can be moved to just before line 170 to avoid some allocations if validation fails.